librad/src/net/gossip.rs

```diff
@@ -111,6 +112,18 @@ where
     pub val: A,
 }
 
+#[derive(Clone, Debug)]
+pub enum MembershipInfo<Addr>
+where
+    Addr: Clone + Eq + Hash,
+{
+    Join {
+        advertised_info: PeerAdvertisement<Addr>,
+        remote_id: PeerId,
```

Comment:
Do you think it is significant to surface this? It's just the last hop of the random walk a `Join` message takes. There might be a case for tracing the entire path, but probably only in a debug build.

Reply:
Don't think so, unless an explicit use-case comes up. Was more mindless mirror of the fields of the lower-level variant.